peps/pep-9999.rst

```diff
@@ -0,0 +1,937 @@
+a ``dep:virtual/`` entry (e.g. ``dep:generic/gcc`` would provide ``dep:virtual/compiler/c``).
+Entries without ``provides`` content or, if populated, only with ``dep:virtual/`` identifiers,
+are considered canonical.
+
+The list of known ecosystems assigns an identifier to each ecosystem and reports the
+canonical location for its mapping. The known package managers are also provided here
+so tools can map package manager names back to ecosystems without querying each mapping.
+
+.. JRG: Not sure if tying the package manager list to the known ecosystems list is worth it.
+
+The mappings specify which ecosystem-specific identifiers provide the canonical entries
+available in the central registry. Its main content is a list of dictionaries, in which each entry consists of:
+
+- an ``id`` field with the ``dep:`` canonical identifier.
+
+- an optional free form ``description`` text.
+
+- a ``specs`` field whose value MUST be one of:
+
+  - a dictionary with three keys (``build``, ``host``, ``run``). The values
+    MUST be a string or list of strings representing the ecosystem-specific package
+    identifiers.
+
+  - for convenience,  a string or a list of strings are also accepted as a
+    shorthand form. In this case, the identifier(s) will be used to populate
+    the three categories mentioned in the item above.
+
+  - an empty list, which is understood as the ecosystem not having packages to
+    provide such dependency.
+
+- a ``specs_from`` field whose value is a ``dep:`` identifier from which the ``specs``
+  field will be imported. Either ``specs`` or ``specs_from`` MUST be present.
+
+- an optional ``urls`` field whose value MUST be a dictionary that maps a string
+  to a URL. This is useful to link to external resources that provide more
+  information about the mapped packages.
+
+Some examples from the conda-forge mapping include:
+
+.. code-block:: js
+
+  [
+    {
+      "id": "dep:generic/zlib",
+      "description": "zlib data compression library for the next generation systems. From zlib-ng/zlib-ng.",
+      "specs": "zlib-ng",  // Simplest form
+      "urls": {
+        "feedstock": "https://github.com/conda-forge/zlib-ng-feedstock"
+      }
+    },
+    {
+      "id": "dep:generic/libwebp",
+      "description": "WebP image library. libwebp-base ships libraries; libwebp ships the binaries.",
+      "specs": {  // expanded form with single spec per category
+        "build": "libwebp-base",
+        "host": "libwebp-base",
+        "run": "libwebp"
+      },
+      "urls": {
+        "feedstock": "https://github.com/conda-forge/libwebp-feedstock"
+      }
+    },
+    {
+      "id": "dep:generic/clang",
+      "description": "Development headers and libraries for Clang",
+      "specs": { // expanded form with specs list
+        "build": [
+          "clang",
+          "clangxx"
+        ],
+        "host": [
+          "clangdev"
+        ],
+        "run": [
```

Comment:
This seems incorrectly specified. `clang` and `clang++` binaries are missing. If I specify `dep:generic/clang` as a runtime dependency for my installed package, changes are I want the compiler rather than (or in addition to) `clang-format` & co.

Reply:
Fixed